neqo-transport/src/connection.rs

```diff
@@ -2259,10 +2259,13 @@ impl Connection {
     }
 
     fn cleanup_streams(&mut self) {
+        self.send_streams.clear_terminal();
+
         let recv_to_remove = self
             .recv_streams
             .iter()
             .filter(|(_, stream)| stream.is_terminal())
+            .filter(|(id, _)| id.is_uni() || self.send_streams.get(**id).is_err())
             .map(|(id, _)| *id)
```

Comment:
Why is this not a single filter_map call?

Reply:
I think it's easier to understand with successive filter steps: first, is this recv stream terminal? And then, is this recv stream either unidirectional, or has its send stream already been cleaned up?